### DISCUSSION TITLE
AssetManager uploadFile not working when adding image via click

### DISCUSSION THREAD

**Discussion by srinivasskeme** [GrapesJS/grapesjs#5189] [Q&A] [answered]
I am using custom upload function to manage asset file upload to server. The event.dataTransfer is not passed when the user clicks on the asset manager to upload files. The functionality works fine when user drops the files on the container.

Additionally, the inline documentation says I will have to emit events `asset:upload:*` if I plan to listen to those events in the rest of the app. I am unable to find the documentation to emit events. [Inline Doc Ref](https://github.com/GrapesJS/grapesjs/blob/047dc35412eaa7d8d53ef634420f93f2fd455f69/src/asset_manager/config/config.ts#L81)

[Reproduced on StackBlitz](https://stackblitz.com/edit/stackblitz-starters-ynilpx?devToolsHeight=33 )

I am using `grapesjs` with `grapesjs-mjml` plugin.

**Comment by artf** [comment, marked as answer]
The example in the doc shows you how to get properly the file
```js
const file = event.dataTransfer?.files[0] || event.target.files[0];
```

**Reply by srinivasskeme** [reply]
Thanks @artf. I missed the example part in docs. Is there any docs on how to emit `asset:upload:*` events during the custom upload process?

**Reply by artf** [reply]
unfortunately right now I can only recommend following the [source method](https://github.com/GrapesJS/grapesjs/blob/ca03f2ca854cdba09f84b0df88f0741a7b7775bd/src/asset_manager/view/FileUploader.ts#L140) and see how during the fetch events are triggered
